splice_machine/src/main/java/com/splicemachine/derby/stream/function/QuoteTrackingTokenizer.java

```diff
@@ -69,211 +140,309 @@
-                    currentColumn.append(c);
                 }
+            } else { // we didn't because the column size is too small. Just reset the size and let SB grow exponentially.
+                currentColumn.setLength(0);
+            }
+        }  // else is a no-op, we don't care at this moment about the columns
+    }
 
-            }else{
-				/*
-				 * QUOTE_MODE (within quotes).
-				 */
-                if(c==quoteChar){
-                    int nextCharIndex=charIndex+1;
-                    boolean availableCharacters=nextCharIndex<line.length();
-                    boolean nextCharIsQuote=availableCharacters && line.charAt(nextCharIndex)==quoteChar;
-                    if(nextCharIsQuote){
-						/*
-						 * An escaped quote (""). Add a single quote, then move the cursor so the next iteration of the
-						 * loop will read the character following the escaped quote.
-						 */
-                        currentColumn.append(c);
-                        charIndex++;
-
-                    }else{
-						/*
-						 * A single quote ("). Update to NORMAL (but don't save quote), then continue to next character.
-						 */
-                        state=TokenizerState.NORMAL;
-                        quoteScopeStartingLine=-1; // reset ready for next multi-line cell
-                    }
-                }else{
-					/*
-					 * Just a normal character, delimiter (they don't count in QUOTESCOPE) or space. Add the character,
-					 * then continue to next character.
-					 */
-                    currentColumn.append(c);
-                }
+    private String getCurrentLine() {
+        if (checkExactSizeFirst) {
+            return currentRow.last();
+        } else if (exactColumnSizes != null) {
+            return currentRow.at(rowIdx);
+        } else {
+            return currentRow.last();
+        }
+    }
+
+    private boolean readFirstLine() throws IOException {
+        String line;
+        do {
+            line = readLine();
+            if (line == null) {
+                return false;
             }
+        }
+        while (ignoreEmptyLines && line.isEmpty() || (commentMatcher != null && commentMatcher.isComment(line)));
+        currentRow.append(line);
+        return true;
+    }
+
+    private void handleNormalState(final List<String> columns, final BooleanList quotedColumns) {
+        if (charIndex == getCurrentLine().length()) { // Newline. Add any required spaces (if surrounding spaces don't need quotes) and return (we've read a line!).
+            finishCol(columns, quotedColumns);
+            state = TokenizerState.FINISHED;
+        } else {
+            final char c = getCurrentLine().charAt(charIndex);
+            if (c == delimeterChar) { // Delimiter. Save the column (trim trailing space if required) then continue to next character.
+                finishCol(columns, quotedColumns);
+                potentialSpaces = 0;
+                allocateColumnMemory();
```

Comment:
I think this call is not needed, we already allocated a StringBuilder big enough for the largest column, the only thing we need is to "clear" the StringBuilder (`.setLength(0)`) right?

Reply:
You're right, no need for resizing.

Reply:
... actually we do need this to reset the `StringBuilder` for the next column, otherwise we keep adding to the `StringBuilder`.